File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/middlewares/validation/schema/threed_schema.js
// Middleware de validation de schéma pour le module threed (Node.js)
function threedSchema(req, res, next) {
  if (req.schema && typeof req.schema.validate === 'function') {
    req.schema.validate(req.body || {});
  }
  next();
}

module.exports = threedSchema;

File: backend/metier_generator/templates/threed/middlewares/validation/schema/threed_schema.py
"""
Middleware de validation de schéma pour le module threed.
Valide un dictionnaire data selon un schéma JSON Schema simplifié.
"""


def validate_schema(schema, data):
    if not schema or not isinstance(schema, dict):
        return False
    # Validation ultra simplifiée pour l’exemple
    if schema.get("type") == "object":
        props = schema.get("properties", {})
        for key, prop in props.items():
            if prop.get("type") == "string" and not isinstance(data.get(key), str):
                return False
    return True
